os/board/rtl8721csm/src/rtl8721csm_boot.c

```diff
@@ -85,6 +91,25 @@ extern FAR struct gpio_lowerhalf_s *amebad_gpio_lowerhalf(u32 pinname, u32 pinmo
 /************************************************************************************
  * Public Functions
  ************************************************************************************/
+#ifdef CONFIG_PRODCONFIG
+static void up_check_prod(char *value)
+{
+	u8 prod_disable;
+	EFUSERead8(0, EFUSE_SEC_CONFIG_ADDR1, &prod_disable, L25EOUTVOLTAGE);
+	if (prod_disable == 0xff) {
+		*value |= SWD_ENABLED;
+		*value |= DOWNLOAD_ENABLED;
```

Comment:
Is this necessary to be bit handling?
This function should be implemented by chipset vendor. It's not good to understand to port.
```
int up_check_prodswd(void);
int up_check_proddownload(void);
{
 return up_check_prod(void);
}
```
It's ok to bit handling in the driver side because chipset vendor doesn't need to know.